pkg/lib/cockpit-components-upload.jsx

```diff
@@ -0,0 +1,157 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import cockpit from "cockpit";
+
+const BLOCK_SIZE = 16 * 1024;
+const TOTAL_FRAME_WINDOW = 128;
+/*
+ * Something smart about upload
+ */
+export class UploadHelper {
+    constructor(destination, onProgress) {
+        this.destination = destination;
+        this.progressCallback = onProgress;
+        this.count = TOTAL_FRAME_WINDOW;
+
+        // Promise resolvers
+        this.resolveClose = null;
+        this.resolveFlush = null;
+        this.resolveHandler = null;
+
+        this.channel = cockpit.channel({
+            binary: true,
+            payload: "fsreplace1",
+            path: this.destination,
+            superuser: "try",
+            "send-acks": "frames"
```

Comment:
This is tricky.  We no longer have a way to acknowledge this, so it means that anyone using this file is going to be implicitly depending on (whatever) cockpit bridge version.  I wonder if we have a way to advertise/enforce that dependency relationship.

Reply:
As far as I know not, the idea was to have `send-acks` back in the ready message if supported.

So we can do that? And then `console.error`, it is not ideal but the best we can do afaik.